### PR TITLE
Add connector (api_outbound_delivery_srv_0002)

### PR DESCRIPTION
**Pull request by osanda2024**
## Purpose
This pull request focuses on adding the first connector (api_outbound_delivery_srv_0002) under the delivery and transportation module of SAP S/4 HANA.

Fixes:

## Checklist
- [ ]  Add first connector
- [ ] Sanitize the Open API Specification
- [ ] Add operation ID to the Open API Specification
- [ ]  Generate client using the Open API Specification
- [ ] Sanitize the generated client
- [ ] Generate mock server
- [ ] Add tests

### CODE REVIEW

File: ballerina/api_outbound_delivery_srv_0002/build.gradle [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+ * under the License.
+ */
+import org.apache.tools.ant.taskdefs.condition.Os
+
+plugins {
+    id 'net.researchgate.release'
+    id 'io.ballerina.plugin'
+}
+
+description = 'Ballerina - SAP S/4HANA Inbound Delivery API'
```
Comment (niveathika) [inline ballerina/api_outbound_delivery_srv_0002/build.gradle]: Description seems to mismatch

File: ballerina/api_outbound_delivery_srv_0002/build.gradle [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+def packageName = "sap.s4hana.api_outbound_delivery_srv_0002"
+def moduleVersion = project.api_outbound_delivery_srv_0002Version.replace("-SNAPSHOT", "")
+def ballerinaTomlFile = file("${project.projectDir}/Ballerina.toml")
+def ballerinaTomlFilePlaceHolder = new File("${project.rootDir}/build-config/resources/BallerinaDelivery.toml")
+
+ballerina {
+    packageOrganization = project.ext.packageOrg
+    module = packageName
+    isConnector = true
+    platform = "any"
```
Comment (niveathika) [inline ballerina/api_outbound_delivery_srv_0002/build.gradle:36]: testcoverage param missing

File: ballerina/api_outbound_delivery_srv_0002/docs.json [hunk 3]
```diff
@@ -0,0 +1,7 @@
+{
+  "description": "The `ballerinax/sap.s4hana.api_del_doc_with_credit_block` package offers APIs for seamless integration with the [Delivery Document with Credit Block API v1.0.0](https://api.sap.com/api/API_DEL_DOC_WITH_CREDIT_BLOCK/overview). The service contains entities for credit blocked delivery document and reject reason. Once the delivery document has been checked, released, or rejected, a success message containing the document number is sent in the response.",
+  "communication-scenario": "Sales Credit Management Integration",
+  "import-statement": "import ballerinax/sap.s4hana.api_del_doc_with_credit_block as delDoc;",
+  "client-init": "delDoc:Client delDocClient = check new (",
+  "api-invocation": "delDoc:CollectionOfA_CreditBlockedDeliveryDocWrapper listcreditBlockedDeliveryDocs = check delDocClient->listA_CreditBlockedDeliveryDocs();"
+}
```
Comment (niveathika) [inline ballerina/api_outbound_delivery_srv_0002/docs.json]: Docs are mismatched.

File: ballerina/build.gradle [hunk 4]
```diff
@@ -24,6 +24,7 @@ description = 'Ballerina - SAP S/4HANA Delivery and Transporation APIs'
 subprojects {
     ext {
         packageOrg = "ballerinax"
+        ballerinaTomlFilePlaceHolder = new File("${project.rootDir}/build-config/resources/Ballerina.toml")
```
Comment (niveathika) [inline ballerina/build.gradle]: No need

**Comment by niveathika** [comment]
@osanda2024 I see some field reordered unnecessarily in the sanitized spec.

**Review by niveathika — CHANGES_REQUESTED** [review]

File: ballerina/build.gradle (hunk 4 above)
Comment (osanda2024) [reply ballerina/build.gradle]: Updated in [Update reviewed changes](https://github.com/ballerina-platform/module-ballerinax-sap.s4hana.delivery/pull/9/commits/2628f3191c53a7f8a567b1d7e2c7035c358ff411)

File: ballerina/api_outbound_delivery_srv_0002/build.gradle (hunk 1 above)
Comment (osanda2024) [reply ballerina/api_outbound_delivery_srv_0002/build.gradle]: Updated in [Update reviewed changes](https://github.com/ballerina-platform/module-ballerinax-sap.s4hana.delivery/pull/9/commits/2628f3191c53a7f8a567b1d7e2c7035c358ff411)

File: ballerina/api_outbound_delivery_srv_0002/docs.json (hunk 3 above)
Comment (osanda2024) [reply ballerina/api_outbound_delivery_srv_0002/docs.json]: Updated in [Add documentation for outbound delivery api](https://github.com/ballerina-platform/module-ballerinax-sap.s4hana.delivery/pull/9/commits/7d7821adbe2dd546848b9a426b306d8509f958b2)

File: ballerina/api_outbound_delivery_srv_0002/build.gradle (hunk 2 above)
Comment (osanda2024) [reply ballerina/api_outbound_delivery_srv_0002/build.gradle:36]: Updated in [Update test coverage parameters](https://github.com/ballerina-platform/module-ballerinax-sap.s4hana.delivery/pull/9/commits/d4040a58998317ac36eeebb70135e08df936b118)

**Review by niveathika — DISMISSED** [review]

**Review by niveathika — APPROVED** [review]

**Pull request merged**